### ISSUE TITLE
Improve error experience when targeting c++/CLI with .NET framework MSBuild on ARM64

### ISSUE THREAD

**Issue by marcpopMSFT** [dotnet/msbuild#5928]
### Issue Description
It is possible though not recommended for customers to install the build tools sku on an ARM64 windows machine in emulation mode.  We support c++/CLI in .NET 5 and support ARM64 in .NET 5 but do not support ARM64 c++/CLI

### Steps to Reproduce
Install VS or build tools on a windows arm64 machine (VS installer will warn you that this is not supported).
Try to build a c++/CLI project

### Expected Behavior
Improve the error experience to clarify that this combination of tools and configuration are not actually supported.

### Actual Behavior
Confusing error

**Comment by rainersigwald** [comment]
Do we know what the current confusing error is? That might help figure out the best place to put the early error.

**Comment by v-zhiyul** [comment]
@marcpopMSFT Could you provide confusing error information such as image to us? Thanks a advance！

**Comment by marcpopMSFT** [comment]
This was filed on behalf of a customer who reported a confusing error so I don't have the specific text handy.  @AaronRobinsonMSFT may have those details. I recommend this get moved into our dev18 bucket as we have some work planned there that may improve this scenario.

**Comment by AaronRobinsonMSFT** [comment]
> @marcpopMSFT Could you provide confusing error information such as image to us? Thanks a advance！

I don't think this is how I would approach the issue. I recommend trying the scenario locally and observing the UX. The error messages if I recall are around missing dependent packages, but it has been quite some time and don't recall specifics.
